clock/Counter.qml

```diff
@@ -0,0 +1,115 @@
+import QtQuick
+
+Item {
+    id: root
+    property var numberType: value
```

Comment:
Mi fa strano che questo componente non abbia una sua dimensione...se lo metti a schermo senza dargliele non ti si vede nulla. Io imposterei delle dimensioni, a mano oppure inserendo tutti i figli dentro un Column che poi userai per prendere le dimensioni. @giannivalenti cosa ne dici?

Reply:
Per quanto riguarda la larghezza, tutti i figli sono ancorati a `parent.left` e `parent.right` e questo è un bene. Però io assegnerei anche all'elemento `root` una larghezza di default che poi eventualmente potrà essere reimpostata da chi lo instanzia.

Per quanto riguarda l'altezza invece la question è un po' più delicata. Se l'elemento `root` fosse piazzato sullo schermo senza impostarne l'altezza succederebbe che tutti i suoi figli verrebbero comunque disegnati, ma solo perché il Qml permette loro di "uscire" dall'elemento `root`. Cioè, l'`Item` avrebbe altezza zero, ma i suoi figli potrebbero comunque uscire fuori dal padre e verrebbero disegnati ugualmente.

In sintesi, sono d'accordo con @lucaotta: sarebbe meglio che l'`Item` `root` avesse delle dimensioni impostate. Se vuoi divertirti a capire cosa succede nei vari casi, puoi provare ad aggiungere `clip: true` per l'`Item` `root` e vedere come cambiano le cose aumentandone via via l'altezza.